File: services/restaurant-service/src/main/java/com/example/restaurantservice/api/response/RestaurantDTO.java
package com.example.restaurantservice.api.response;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;
import lombok.Builder;

import java.util.Set;



@Builder

public record RestaurantDTO(
        String restaurantName,
        @Pattern(regexp = "^[+]\\d{2}\\s\\d{3}\\s\\d{3}\\s\\d{3}$")
        String phone,
        @Email
        String email,
        String address,
        String ownerEmail,
        Set<RestaurantStreetDTO> restaurantStreets
) {


}

File: FoodOrderingApp/src/models/restaurant-dto.ts
import {RestaurantStreetDTO} from './restaurant-street-dto';

export interface RestaurantDTO {
  restaurantId: string;
  restaurantName: string;
  phone: string;
  email: string;
  address: string;
  ownerEmail: string;
  restaurantStreets: RestaurantStreetDTO[];
}
